rct229/rules/section5/section5rule24.py

```diff
@@ -0,0 +1,259 @@
+from rct229.data_fns.table_G3_4_fns import table_G34_lookup
+from rct229.rule_engine.rule_base import (
+    RuleDefinitionBase,
+    RuleDefinitionListIndexedBase,
+)
+from rct229.rule_engine.user_baseline_proposed_vals import UserBaselineProposedVals
+from rct229.ruleset_functions.get_building_scc_window_wall_ratios_dict import (
+    get_building_scc_window_wall_ratios_dict,
+)
+from rct229.ruleset_functions.get_opaque_surface_type import OpaqueSurfaceType as OST
+from rct229.ruleset_functions.get_opaque_surface_type import get_opaque_surface_type
+from rct229.ruleset_functions.get_surface_conditioning_category_dict import (
+    SurfaceConditioningCategory as SCC,
+)
+from rct229.ruleset_functions.get_surface_conditioning_category_dict import (
+    get_surface_conditioning_category_dict,
+)
+from rct229.utils.jsonpath_utils import find_all
+from rct229.utils.std_comparisons import std_equal
+
+DOOR = schema_enums["SubsurfaceClassificationType"].DOOR.name
+
+
+class Section5Rule24(RuleDefinitionListIndexedBase):
+    """Rule 24 of ASHRAE 90.1-2019 Appendix G Section 5 (Envelope)"""
+
+    def __init__(self):
+        super(Section5Rule24, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, False),
+            required_fields={
+                "$": ["weather"],
+                "weather": ["climate_zone"],
+            },
+            each_rule=Section5Rule24.BuildingRule(),
+            index_rmr="baseline",
+            id="5-24",
+            description="Vertical fenestration U-factors for residential, non-residential and semi-heated spaces in the baseline model must match the appropriate requirements in Table G3.4-1 through G3.4-8 for the appropriate WWR in the baseline RMD",
+            list_path="ruleset_model_instances[0].buildings[*]",
+        )
+
+    def create_data(self, context, data=None):
+        rmr_baseline = context.baseline
+        climate = rmr_baseline["weather"]["climate_zone"]
+
+        # TODO It is determined later we will modify this function to RMD level -
+        # The implementation is temporary
+        bldg_scc_wwr_ratio_dict = {}
+        for building_b in find_all(self.list_path, rmr_baseline):
+            bldg_scc_wwr_ratio_dict[
+                building_b["id"]
+            ] = get_building_scc_window_wall_ratios_dict(climate, building_b)
+
+        return {
+            "climate_zone": rmr_baseline["weather"]["climate_zone"],
+            "bldg_scc_wwr_ratio_dict": bldg_scc_wwr_ratio_dict,
+        }
+
+    class BuildingRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section5Rule24.BuildingRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+                each_rule=Section5Rule24.BuildingRule.AboveGradeWallRule(),
+                index_rmr="baseline",
+                list_path="$..surfaces[*]",
+            )
+
+        def manual_check_required(self, context, calc_vals=None, data=None):
```

Comment:
By my reading of the RDS, we need to set a `manual_review_flag` in `create_data()` rather than have a `manual_check_required()` method. The manual check is required at the subsurface level rather than the building level.